### ISSUE TITLE
Replace all the `*Type` classes by enumerations

### ISSUE THREAD

**Issue by meritoo** [meritoo/common-library#13]
TL;DR
- PHP enumerations: https://www.php.net/manual/en/language.types.enumerations.php
- All the classes that extend `Meritoo\Common\Type\Base\BaseType`

**Aa example of new `enum`:**

```php
<?php

declare(strict_types=1);

namespace Meritoo\Common\Enums;

enum OopVisibility: string
{
  case Private = '3';
  case Protected = '2';
  case Public = '1';
}
```

**A new piece of `[CHANGELOG.md](CHANGELOG.md)`:**

6. Replace all the `*Type` classes
 with [enumerations](https://www.php.net/manual/en/language.types.enumerations.php) (
 classes that extend `Meritoo\Common\Type\Base\BaseType`)

 | Before                    | After                |
 |---------------------------|----------------------|
 | `class OopVisibilityType` | `enum OopVisibility` |
 | ...                       | ...                  |
 | ...                       | ...                  |
